pulp_rpm/app/tasks/publishing.py

```diff
@@ -246,14 +241,17 @@ def create_repomd_xml(packages, publication, extra_repomdrecords, sub_folder=Non
     oth_db = cr.OtherSqlite(oth_db_path)
     upd_xml = cr.UpdateInfoXmlFile(upd_xml_path)
 
-    pri_xml.set_num_of_pkgs(len(packages))
-    fil_xml.set_num_of_pkgs(len(packages))
-    oth_xml.set_num_of_pkgs(len(packages))
+    packages = Package.objects.filter(pk__in=contents)
+    total_packages = packages.count()
+
+    pri_xml.set_num_of_pkgs(total_packages)
+    fil_xml.set_num_of_pkgs(total_packages)
+    oth_xml.set_num_of_pkgs(total_packages)
 
     # Process all packages
-    for package in packages:
+    for package in packages.iterator():
```

Comment:
By default, [iterator](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#iterator) only bring 2000 rows into memory, as we only would use packages on this loop, it was better to use iterator than caching the entire query.